src/infrastructure/entities/AddedCoin.ts

```diff
@@ -0,0 +1,32 @@
+import { Entity, PrimaryColumn, Column, Index } from "typeorm";
+import { getBinaryType } from "./OrmAnnotationTypes";
+
+@Entity()
+@Index(["addressId"])
+@Index(["coinStatus"])
+@Index(["assetId"])
+export class AddedCoin {
+  @PrimaryColumn({ type: 'text' })
+  addressId!: string;
+
+  @PrimaryColumn({ type: getBinaryType() })
+  coinId!: Buffer;
+
+  @Column({ type: getBinaryType() })
+  parentCoinInfo!: Buffer;
+
+  @Column({ type: getBinaryType() })
+  puzzleHash!: Buffer;
+
+  @Column({ type: 'text' })
+  amount!: string;
+
+  @Column({ type: 'integer' })
+  syncedHeight!: number;
+
+  @Column({ type: 'text' })
+  coinStatus!: string;
```

Comment:
Maybe we can monitor the mempool in the future. For now, we should track pending coins in the db and to determine if the pending state is still active we can add a pending_expiring date. Maybe refresh on every new block.

Insertting a bloc should be a single transaction. Ideally make a trigger that is activated at the end of the block.